### PR TITLE
New commit guidelines

### PR DESCRIPTION
**Pull request by htgoebel**
This is a draft for new commit guidelines. This emphasis on
- clean history
- good commit messages
- cleaning up commit history prior to push.

(The rearrangement of the text is less relevant.)

@pyinstaller/core-developers @pyinstaller/contributors @pyinstaller/hook-maintainers @pyinstaller/issue-contributors Your comments are welcome!

### CODE REVIEW

File: doc/development/commit-messages.rst [hunk 1]
```diff
@@ -1,3 +1,76 @@
...
+
+A commit should be one (and just one) logical unit.
+It should be something that someone might want to patch or
+revert in its entirety, and never piece-wise.
+If it could be useful in pieces, make separate commits.
+
+* Make small patches (i.e. work in consistent increments).
+
+* Reformatting code without functional changes will generally not be
+  accepted (for rational see :issue:`2727`).
```
Comment (bjones1) [inline doc/development/commit-messages.rst]: rationale, not rational.

File: doc/development/commit-messages.rst [hunk 2]
```diff
@@ -1,3 +1,76 @@
...
+  If such changes are required, separate it into a commit of it own
+  and document as such.
```
Comment (bjones1) [inline doc/development/commit-messages.rst]: its own, not it own.

**Comment by Suzumizaki** [comment]
How to do when Appveyor or Travis-CI fails but the committer seems not responsible? :)

File: doc/development/commit-messages.rst (hunk 1 above)
Comment (htgoebel) [reply doc/development/commit-messages.rst]: Thanks for spotting the typos.

**Pull request merged**